### PR TITLE
Header component & assets folder

### PR DESCRIPTION
**Pull request by Fraasi**
Ok, here's the basics of the header.
There was quite a lot else I did here, see if I remember them:
* header component. Need of course links later, react router?
* added assets folder with fonts, images & I ported all css variables from sass to regular css and load it and font in the main index.css file so they should be usable in all the components throughout the app.
* I removed create-react-app stuff, and added header to the page, so running `npm start` you can actually see something.
* found a bug in css. [Css variables do not work inside mediaqueries](https://stackoverflow.com/questions/40722882/css-native-variables-not-working-in-media-queries), +1 for the sass team 😢  This is going to make the mobile dev a tad bit more tedius. But then again, now you can easily see the value you need quickly from variables.css.
* I think it's more clearer if the image isn't clickable for the home page, but there is separate 'home'-button at the top, thoughts?
* added header to storybook, component doesn't seem to render perfectly there, compare pics below.
* propably forgot some little thing.

### CODE REVIEW

**Comment by Fraasi** [comment]
Forget  
package-lock after npm install &
Merge remote-tracking branch 'upstream/master'  
I forgot to push after merging your changes to my local repo, so now they are part of this. Much to learn how to git still there is.

File: src/assets/variables.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  --mobile: 550px;
+
+
+  /* Animations */;
+  --timing-function: cubic-bezier(0.4, 0.0, 0.2, 1);
+
+
+  /* Misc */;
+  --standard-border-radius: 5px;
+}
```
Comment (meiamsome) [inline src/assets/variables.css:37]: This should have a new line here

**Comment by meiamsome** [comment]
I think we should drop the `package-lock.json` changes, they are not important in this case (It's because we're on different operating systems that there's any difference at all.) We can drop the merge commit after that too.

You're right to suspect that the Storybook isn't capturing global CSS, it's kind of not the point of Storybook - the idea is that the components shouldn't rely on there being any parent styles leaking down to work correctly, to help enforce the component design pattern. It's a strange one when there's theming involved, I'm not sure 

I think the lack of `var` support for media-queries & the fact that there's [~13% of web users](https://caniuse.com/#feat=css-variables) who have no support for CSS variables kinda kills the approach of using them exclusively for me in the end, not to mention that it makes the components dependent on external CSS (Which is why the Storybook renders incorrectly). That said, I am happy to continue down this route for a while, even if the uncentralised media queries are a pain.

**Review by meiamsome — CHANGES_REQUESTED** [review]
See previous comments

**Comment by Fraasi** [comment]
> We can drop the merge commit after that too.  

I didn't know that's possible, how would one do it? I'm thinking something in [this](https://blog.github.com/2015-06-08-how-to-undo-almost-anything-with-git/) article will apply to this situation,..  

Yeah, I'm starting to get the storybook 'totally independent' component-thing.  

> That said, I am happy to continue down this route for a while.  

Me too, for now, but if more problems arise down the road, we can always rethink this and try to find a better approach to styling... maybe styled-components 😃 

I' will be quite busy today, but I'll read that article and find away to get rid of those unrelated commits when I get the chance.

**Comment by Fraasi** [comment]
Whoops, seems I got something wrong. I tried to delete package-lock, and now it's the only one there. 
I'll get some morning coffee and get back to reading...

**Comment by meiamsome** [comment]
My suggestion would be to do the following: 

```
git reset --hard  95dabc4 # gets us back to where we were before
git rebase -i upstream/master # start an interactive rebase from the upstream/master branch (Replace with a branch that is inline with the master in this repo, you could also use the hash 2301d07)
```
This will bring up an editor something like this (Note that the commits brought in from the merge have been included individually, in this case they should all be dropped):
```
pick 873d004 package-lock after npm install
pick 9d96316 Add some simple types
pick 3aba9e2 Add storybook
pick 4790625 Add VideoCard element for testing
pick 94863b3 Header component, assets folder
pick 95dabc4 added coding train favicon and title
```

Where you can chose your plan of action:

```
d 873d004 package-lock after npm install
d 9d96316 Add some simple types
d 3aba9e2 Add storybook
d 4790625 Add VideoCard element for testing
p 94863b3 Header component, assets folder
p 95dabc4 added coding train favicon and title
```
Drop everything, except the header, favicon and title.
Then `git log` should show something like:
```
commit 6a99af54421155f0a0d030d081b2243a69c351f9 (HEAD -> header)
Author: Fraasi <>
Date:   Sun Jan 6 14:31:50 2019 +0200

    added coding train favicon and title

commit a644c77f993fe9ece9020d642c2d815581db2657
Author: Fraasi <>
Date:   Sun Jan 6 13:26:00 2019 +0200

    Header component, assets folder

commit d3429cf8af52e6e5be2bbd80706a1c728ead17ef (upstream/master, master)
Author: Oliver Wright <meiamsome@meiamso.me>
Date:   Tue Jan 1 01:22:28 2019 +0000

    Add .tslint to airbnb standard
```

Hope that helps you

**Comment by Fraasi** [comment]
Thank you, the reset command was a 'life saver'-medal worth 🥇 Now under the original pull request comment there's nothing... but under the commits tab everything seems to be as it should be...  
but now circleci is not passing my favicon change... what's up with that? I can't tell what it is that it is complaining about.   
Edit: under npm run lint
>> tslint src/**/*.tsx
Warning: The 'no-boolean-literal-compare' rule requires type information.
ERROR: src/components/Header/index.test.tsx:5:1 - Consecutive blank lines are forbidden
ERROR: src/components/Header/index.tsx:4:1 - Consecutive blank lines are forbidden

Which is strange because the favicon commit didn't ouch any .tsx-files
🤔

**Comment by Fraasi** [comment]
@meiamsome Thank you for your help! 🤗 Hopefully everything is in order now?

**Comment by meiamsome** [comment]
> Now under the original pull request comment there's nothing... but under the commits tab everything seems to be as it should be...

Yep, They moved past the comments in the comment chain to where they were last commited chronologically (Despite them still having the original date - each commit has a date that the changes were made, and a date when it was commited that can be different)

> Which is strange because the favicon commit didn't ouch any .tsx-files
> :thinking:

Yeah, it would fail the previous commit too, but it will only run for the final commit on a branch for efficiency reasons, if that makes sense?

I'll have a look through again in a bit :)

**Comment by Fraasi** [comment]
>  if that makes sense?

Nothing makes sense at the moment :weary: I need a break from computing & I have a thousand other things to do today... Learning about rebase was a bit too much first thing in the morning,

**Review by meiamsome — APPROVED** [review]

**Pull request merged**